src/sync/mod.rs

```diff
@@ -70,34 +121,29 @@ pub async fn begin_sync(
                 _ => return Err(InternalNonLocalPendingCommit),
             }
         }
-        // Contract is that BatchPushInfo is present if we attempted to push.
-        begin_sync_resp.sync_info.batch_push_info = Some(BatchPushInfo {
-            http_status_code: 0,
-            error_message: str!(""),
-        });
         let push_req = push::BatchPushRequest {
-            client_id: client_id.clone(),
+            client_id,
             mutations: push_mutations,
         };
         debug!(lc, "Starting push...");
         let push_timer = rlog::Timer::new().map_err(InternalTimerError)?;
         let push_resp = pusher
             .push(
                 &push_req,
-                &begin_sync_req.batch_push_url,
-                &begin_sync_req.data_layer_auth,
-                &sync_id,
+                &req.batch_push_url,
+                &req.data_layer_auth,
+                sync_id,
             )
             .await;
         // Note: no map_err(). A failed push does not fail sync, but we
```

Comment:
I think we should reconsider this, but I guess we would need a way to signal potentially temporary errors from fatal errors to the caller. Like if there is no main head that is fatal. But if the http request timed out that's temporary. We do not have a good way to do this yet :(